etcdctl/ctlv3/command/member_command.go

```diff
@@ -257,7 +257,37 @@ func memberPromoteCommandFunc(cmd *cobra.Command, args []string) {
 	}
 
 	ctx, cancel := commandCtx(cmd)
-	resp, err := mustClientFromCmd(cmd).MemberPromote(ctx, id)
+	eps, err := endpointsFromCmd(cmd)
+	if err != nil {
+		ExitWithError(ExitError, err)
+	}
+
+	var (
+		leaderEp    string
+		leaderFound bool
+	)
+	for _, ep := range eps {
```

Comment:
If ctx has a deadline, loop multiple times over the endpoints until the deadline expires or a leader is found? This would reduce the odd of a leader change causing this operation to fail.

Reply:
Ignore if we can do a server side request foward like mentioned in https://github.com/jingyih/etcd/pull/28#issuecomment-487293653